src/main/java/com/mitchej123/hodgepodge/mixins/automagy/MixinItemBlockThirstyTank.java

```diff
@@ -0,0 +1,77 @@
+package com.mitchej123.hodgepodge.mixins.automagy;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.IFluidContainerItem;
+import org.spongepowered.asm.mixin.Mixin;
+import tuhljin.automagy.blocks.ItemBlockThirstyTank;
+import tuhljin.automagy.tiles.ModTileEntity;
+import tuhljin.automagy.tiles.TileEntityThirstyTank;
+
+import java.util.Arrays;
+
+@Mixin(ItemBlockThirstyTank.class)
+public class MixinItemBlockThirstyTank implements IFluidContainerItem {
+    @Override
+    public FluidStack getFluid(ItemStack container) {
+        return container.hasTagCompound() ? FluidStack.loadFluidStackFromNBT(container.stackTagCompound) : null;
+    }
+
+    @Override
+    public int getCapacity(ItemStack container) {
+        int[] glyphs = ModTileEntity.getIntArrayFromNbtOrDefault(container.stackTagCompound, "Glyphs", 0, 6);
+        int glyphOfTheReservoirId = 8;
+        final int glyphOfTheReservoirCount = (int) Arrays.stream(glyphs).filter(glyph -> glyph == glyphOfTheReservoirId).count();
+        return 1000 * (16 + glyphOfTheReservoirCount * TileEntityThirstyTank.CAPACITY_IN_BUCKETS_PER_UPGRADE);
```

Comment:
You might as well use `TileEntityThirstyTank.CAPACITY_IN_BUCKETS_DEFAULT` instead of 16 here